src/Google.Pubsub.V1/SubscriberClient.cs

```diff
@@ -0,0 +1,661 @@
+﻿/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ * Use of this source code is governed by a BSD-style
+ * license that can be found in the LICENSE file or at
+ * https://developers.google.com/open-source/licenses/bsd
+ */
+
+// Generated code. DO NOT EDIT!
+
+using Google.Api.Gax;
+using Google.Apis.Auth.OAuth2;
+using Grpc.Auth;
+using Grpc.Core;
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Google.Pubsub.V1
+{
+    /// <summary>
+    /// Extension methods to assist with using <see cref="SubscriberClient"/>.
+    /// </summary>
+    public static partial class SubscriberExtensions
+    {
+        /// <summary>
+        /// Wrap a GRPC Publisher client for more convinient use.
+        /// </summary>
+        /// <param name="grpcClient">A GRPC client to wrap.</param>
+        /// <param name="settings">
+        /// An optional <see cref="SubscriberSettings"/> to configure this wrapper.
+        /// If null or not specified, then the default settings are used.
+        /// </param>
+        /// <returns>A <see cref="SubscriberSettings"/> that wraps the specified GRPC client.</returns>
+        public static SubscriberClient ToClient(
```

Comment:
We need to have another round of thinking about naming - definitely odd that SubscriberClient doesn't implement ISubscriberClient.